pyproject.toml

```diff
@@ -68,7 +68,7 @@ lz4 = { version = ">=3.1.3", optional = true }
 protobuf = { version = "^4.25", optional = true }
 protoc-gen-openapiv2 = {version = "^0.0.1", optional = true }
 pinecone-plugin-interface = "^0.0.7"
-pinecone-plugin-inference = ">=1.0.2"
+pinecone-plugin-inference = ">=1.0.3"
```

Comment:
Can we change it to the caret style ^1.0.3? That will restrict it to ">=1.0.3, <2.0.0". If inference is doing a major version upgrade, then it should probably result in a major version upgrade for the main client